code.js

```diff
@@ -723,3 +723,32 @@ function addEventListeners() {
     }, 0);
   });
 }
+
+// Creating custom checkbox.
+const all_main_checkboxes = document.querySelectorAll("input[type='checkbox']");
+all_main_checkboxes.forEach(checkbox => {
+  const custom_checkbox = document.createElement("span");
+  const check_icon = document.createElement("img");
+
+  custom_checkbox.className = "custom-checkbox";
+  custom_checkbox.id = checkbox.id;
+  check_icon.className = "check-icon";
+  check_icon.setAttribute("src", "/img/check.svg")
+  
+  custom_checkbox.appendChild(check_icon);
+  checkbox.after(custom_checkbox);
+
+  custom_checkbox.addEventListener("click", (event) => {
+    const main_checkbox = document.querySelector(`input#${event.target.id}`);
+    const is_checked = main_checkbox.hasAttribute("checked");
+    if (!is_checked && event.target.id === checkbox.id) {
+      // check
+      main_checkbox.setAttribute("checked", "");
+      custom_checkbox.classList.add("active-checkbox");
+    } else {
+      // uncheck
+      main_checkbox.removeAttribute("checked");
```

Comment:
here is okay to check or uncheck (746 and 750, but then in registration_api.js we add onClick => what's with that click event?)

Reply:
this place should suport checking/unchecking and inside consumers (registration_api.js) we only send events to xstate etc.

Reply:
in the registration api, it pre-checks checkboxes when we fetch the data from editing the app. So that is a different case.